docs/source/workflows/design_spaces.rst

```diff
@@ -158,3 +155,84 @@ The example code below creates a registers a design space based on this Gem Tabl
     )
 
     registered_design_space = project.design_spaces.register(design_space)
+
+Formulation Design Space
+------------------------
+
+A formulation design space defines the set of formulations that can be produced from a given set of ingredient names, labels and constraints.
+Ingredient names are specified as a set of strings, where each string is a unique ingredient name, e.g., ``{'water', 'salt'}`` to a design space with two ingredients.
+Labels are specified as a mapping from each label to a set of ingredient names that should be given that label is present in a formulation, e.g., ``{'solute': {'salt'}}``.
+An ingredient may be given multiple labels, and these labels are static.
+An ingredient will always be given all applicable labels when present in a formulation.
+
+Constraints restrict the total number or fractional amount of ingredients in formulations sampled from the design space.
+There are three constraints that can be specified as part of a formulation design space:
+
+- :class:`~citrine.informatics.constraints.ingredient_count_constraint.IngredientCountConstraint` constrains the total number of ingredients in a formulation.
+  At least one ingredient count constraint that constrains the total number of ingredients in formulations emitted from the space is required.
```

Comment:
```suggestion
  At least one constraint on the total number of ingredients is required.
```
Trying to make this as clear and succinct as possible, since it's such an important point and likely to cause user-frustration if they miss it.